### DISCUSSION TITLE
Event Triggered: user_ban_failed_login IP

### DISCUSSION THREAD

**Discussion by im-L0r3** [e107inc/e107#4773] [General]
When I receive an email from e107 about "Event Triggered: user_ban_failed_login" the IPv6 address is shown like the following, but when I look at the logs in e107 is shows the IPv4 address. Im blocking IPv6 completely at an upstream firewall, and IPv6 is disabled on the NIC of the Apache2 server.

EMAIL
```
Array
(
    [time] => 1652899707
    [ip] => 0000:0000:0000:0000:0000:ffff:05bc:3e8c
    [other] => 
)
```

BAN LOG FROM SAME EVENT
```
5.188.62.140 | Login failure | 18 May 2022 | Never | Auto-ban: More than 10 failed login attempts
```

**Comment by Deltik** [comment]
The IPv6 address range `::FFFF:0:0/96` is for [IPv4-mapped addresses](https://datatracker.ietf.org/doc/html/rfc5156#section-2.2).  e107 internally stores IPv4 addresses as IPv4-mapped IPv6 addresses.

e107 could improve by also [`print_a()`-ing](https://github.com/e107inc/e107/blob/6f34dc9f29fc48d63f6f11a180b5deafcf2cd438/e107_handlers/notify_class.php#L113-L114) the IPv4 address in the familiar format as well if it is an IPv4-mapped address.  The place for that in the code would probably be [here](https://github.com/e107inc/e107/blob/6f34dc9f29fc48d63f6f11a180b5deafcf2cd438/e107_handlers/login.php#L674-L674).
